### DISCUSSION TITLE
Layers hierarchy

### DISCUSSION THREAD

**Discussion by novhack** [Orama-Interactive/Pixelorama#607] [Ideas]
It would be nice to have a folders/groups in the Timeline.
These folders could be also added to Export dialog.

This is how the feature looks in Aseprite:
![obrazek](https://user-images.githubusercontent.com/42614907/144766031-2b42b2ed-7f63-46c7-a232-cc52b18f27a0.png)

In the future this feature could be potentialy used to group specialised layers. Like layers specific to tilesets or normal maps.

**Comment by mrtripie** [comment]
I'm wondering how these should affect the rendering of the project. Should these strictly be organizational, or would it be better for groups to be their own layers, which blend their children together to get their result. For example, in Krita if you lower the opacity individually on overlapping layers the opacity will combine when overlapping showing the different layers:
<img src="https://user-images.githubusercontent.com/65431647/146622955-e7c05ab2-d99c-4e4e-85ad-795a7bc25ace.png" width=20% height=20%>
However, if you lower the opacity of a group layer which contains those layers, it looks as if the group is a single layer, which can be useful for certian effects:
<img src="https://user-images.githubusercontent.com/65431647/146622975-4a49e16b-146f-46e0-9821-575558c7c19e.png" width=20% height=20%>

If the FX Layers #619 idea is implemented as groups, it would make sense if they worked like that latter (maybe with an option to apply the effect on each layer as it blends them, or just the final result of blending those layers).

I also suggested with the FX Layers that they would store their effects baked into the .pxo file so they could be imported into other software such as game engines easily. Regular groups wouldn't add much complexity to importing, so storing the results of a regular group layer would be a bit wasteful. Would it be better to go with having Groups and FX Groups working consistently or trying to be more efficient with file size by doing something like separating the two ideas with different naming conventions such as "Folders" and "FX Groups". Though this would likely add some complexity to the file format.

**Comment by OverloadedOrama** [comment]
Layer groups are now implemented in #698.
